tools/src/main/scala/caliban/tools/ClientWriter.scala

```diff
@@ -597,9 +597,17 @@ object ClientWriter {
         """
 
   def safeUnapplyName(name: String): String =
-    if (reservedKeywords.contains(name) || name.endsWith("_")) s"$name$$"
+    if (reservedKeywords.contains(name) || name.endsWith("_") || name.charAt(0).isUpper) s"${decapitalize(name)}$$"
```

Comment:
Can you check for size, so that if `name` is empty, it doesn't throw an exception?

Reply:
Was assuming that name can't be empty, as it's a graphql field, but no problem made the change!